go/vt/vttablet/tabletmanager/rpc_actions.go

```diff
@@ -78,6 +78,9 @@ func (agent *ActionAgent) ChangeType(ctx context.Context, tabletType topodatapb.
 		return fmt.Errorf("fixSemiSyncAndReplication failed, may not ack correctly: %v", err)
 	}
 
+	// Let's populate Metadata values.
```

Comment:
I think we should do this in `changeCallback()`, since what we really want is to update the local metadata any time the tablet record changes. This `ChangeType` RPC is only one way that the tablet record might change; the record might get updated externally, and then the RefreshTablet RPC gets called.

With that said, I'm concerned that this change (either in its current form, or with my recommended adjustment) may result in creating these tables in clusters that previously could be sure they didn't need to worry about them (e.g. YouTube). Previously, `populateMetadataTables()` was only ever called if `-restore_from_backup` was enabled. This function is of special concern because it writes directly to non-master MySQL instances (with binlogging disabled on the session). We don't want those tables to unexpectedly start appearing for users that never had them.

A quick fix might be to only update the local metadata if `*restoreFromBackup` (the flag variable in `tabletmanager`) is `true`. This feels weird, but it preserves backward compatibility; a comment would definitely be warranted to explain why you're checking that flag there.

@alainjobart WDYT?

Reply:
I agree we should test the flag.

Reply:
@enisoc  I originally modified it in change Callback, but test case did not go through.  so I change this version, but still worse.  I'll update another version when test case is ok.